packages/core/upload/server/services/upload.js

```diff
@@ -173,39 +173,43 @@ module.exports = ({ strapi }) => ({
       isOptimizableImage,
     } = getService('image-manipulation');
 
-    // Store width and height of the original image
-    const { width, height } = await getDimensions(fileData);
+    try {
+      // Store width and height of the original image
+      const { width, height } = await getDimensions(fileData);
 
-    // Make sure this is assigned before calling upload
-    // That way it can mutate the width and height
-    _.assign(fileData, {
-      width,
-      height,
-    });
+      // Make sure this is assigned before calling upload
+      // That way it can mutate the width and height
+      _.assign(fileData, {
+        width,
+        height,
+      });
 
-    // Upload image
-    await getService('provider').upload(fileData);
+      // Upload image
+      await getService('provider').upload(fileData);
 
-    // Generate thumbnail and responsive formats
-    if (await isOptimizableImage(fileData)) {
-      const thumbnailFile = await generateThumbnail(fileData);
-      if (thumbnailFile) {
-        await getService('provider').upload(thumbnailFile);
-        _.set(fileData, 'formats.thumbnail', thumbnailFile);
-      }
+      // Generate thumbnail and responsive formats
+      if (await isOptimizableImage(fileData)) {
+        const thumbnailFile = await generateThumbnail(fileData);
+        if (thumbnailFile) {
+          await getService('provider').upload(thumbnailFile);
+          _.set(fileData, 'formats.thumbnail', thumbnailFile);
+        }
 
-      const formats = await generateResponsiveFormats(fileData);
-      if (Array.isArray(formats) && formats.length > 0) {
-        for (const format of formats) {
-          if (!format) continue;
+        const formats = await generateResponsiveFormats(fileData);
+        if (Array.isArray(formats) && formats.length > 0) {
+          for (const format of formats) {
+            if (!format) continue;
 
-          const { key, file } = format;
+            const { key, file } = format;
 
-          await getService('provider').upload(file);
+            await getService('provider').upload(file);
 
-          _.set(fileData, ['formats', key], file);
+            _.set(fileData, ['formats', key], file);
+          }
         }
       }
+    } catch (err) {
+      throw new ForbiddenError('File is not a valid image');
```

Comment:
What do think of using `ApplicationError` instead? It feels to me more like a bad request (corrupted data) than a forbidden error.
Also there is the 415 or 422 error code that can work. (If you decide to go this way you would need to modify `packages/core/utils/lib/errors.js` and `packages/core/strapi/lib/services/errors.js`).
The easiest is probably just `ApplicationError` however.

Reply:
Make sense using `ApplicationError`!
I will look at the errors.js files to understand how they work , but I think I will go with ApplicationError